packages/compiler/src/testing/test-host-v2.ts

```diff
@@ -0,0 +1,257 @@
+import { readFile } from "fs/promises";
+import { compilerAssert } from "../core/diagnostics.js";
+import { NodeHost } from "../core/node-host.js";
+import { CompilerOptions } from "../core/options.js";
+import { getRelativePathFromDirectory, joinPaths, resolvePath } from "../core/path-utils.js";
+import { Program, compile as coreCompile } from "../core/program.js";
+import { createSourceLoader } from "../core/source-loader.js";
+import { Diagnostic, NoTarget, SourceFile, StringLiteral, Type } from "../core/types.js";
+import { expectDiagnosticEmpty } from "./expect.js";
+import { StandardTestLibrary, createTestFileSystem } from "./test-host.js";
+import { resolveVirtualPath } from "./test-utils.js";
+import { TestFileSystem } from "./types.js";
+
+// Need a way to combine that with `program`
+export type TestCompileResult = Record<string, Type>;
```

Comment:
This one is one that is a bit of a blocker.

Right now the test framwork expect this result but with teh new approach that would not require creating a test host everytime ahead we also need to be able to get the program another way.

```ts
const host = Tester.createInstance();
const {Foo} = host.compile();
host.program // can now use later
```

standalone?

```ts
// This one use intersection to do this, this kinda break the `as {Foo: Model}` flow used all over the place unforuntatenly.
const {Foo, program} = Tester.compile();

// Nest the types, this makes it a bit more verbose
const {program, types: {Foo} } = Tester.compile();
```